support/cas-server-support-oidc-core-api/src/main/java/org/apereo/cas/oidc/slo/OidcSingleLogoutMessageCreator.java

```diff
@@ -0,0 +1,65 @@
+package org.apereo.cas.oidc.slo;
+
+import org.apereo.cas.logout.slo.SingleLogoutMessage;
+import org.apereo.cas.logout.slo.SingleLogoutMessageCreator;
+import org.apereo.cas.logout.slo.SingleLogoutRequest;
+import org.apereo.cas.oidc.OidcConstants;
+import org.apereo.cas.services.OidcRegisteredService;
+import org.apereo.cas.services.RegisteredServiceLogoutType;
+import org.apereo.cas.support.oauth.web.endpoints.OAuth20ConfigurationContext;
+import org.apereo.cas.util.DigestUtils;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.apache.commons.lang3.StringUtils;
+import org.jose4j.jwt.JwtClaims;
+
+import java.util.HashMap;
+import java.util.UUID;
+
+/**
+ * The message creator for the OIDC protocol.
+ *
+ * @author Jerome LELEU
+ * @since 6.2.0
+ */
+@Slf4j
+@RequiredArgsConstructor
+public class OidcSingleLogoutMessageCreator implements SingleLogoutMessageCreator {
```

Comment:
Please add tests.

Reply:
Sure. But I was waiting for the feedbacks before writing the tests in case I needed to change everything.